index.go

```diff
@@ -126,6 +139,99 @@ func (idx *faissIndex) Add(x []float32) error {
 	return nil
 }
 
+func (idx *faissIndex) ObtainClusterToVecIDsFromIVFIndex() (map[int64][]int64, error) {
+	// This type assertion is required to determine whether to invoke
+	// ObtainClustersWithDistancesFromIVFIndex, SearchClustersFromIVFIndex or not.
+	if ivfIdx := C.faiss_IndexIVF_cast(idx.cPtr()); ivfIdx == nil {
+		return nil, nil
+	}
+
+	clusterVectorIDMap := make(map[int64][]int64)
+
+	nlist := C.faiss_IndexIVF_nlist(idx.idx)
+	for i := 0; i < int(nlist); i++ {
+		list_size := C.faiss_IndexIVF_get_list_size(idx.idx, C.size_t(i))
+		invlist := make([]int64, list_size)
+		C.faiss_IndexIVF_invlists_get_ids(idx.idx, C.size_t(i), (*C.idx_t)(&invlist[0]))
+		clusterVectorIDMap[int64(i)] = invlist
+	}
+
+	return clusterVectorIDMap, nil
+}
+
+func (idx *faissIndex) ObtainClustersWithDistancesFromIVFIndex(x []float32, centroidIDs []int64) (
```

Comment:
@metonymic-smokey I've some questions around why we're passing in an array of centroids and getting back an array of centroids again here, but we can tackle these separately.